plugin.video.retrospect/resources/libs/envcontroller.py

```diff
@@ -0,0 +1,433 @@
+#===============================================================================
+# LICENSE Retrospect-Framework - CC BY-NC-ND
+#===============================================================================
+# This work is licenced under the Creative Commons
+# Attribution-Non-Commercial-No Derivative Works 3.0 Unported License. To view a
+# copy of this licence, visit http://creativecommons.org/licenses/by-nc-nd/3.0/
+# or send a letter to Creative Commons, 171 Second Street, Suite 300,
+# San Francisco, California 94105, USA.
+#===============================================================================
+import os
+import sys
+import time
+import struct
+
+import xbmc
+import xbmcaddon
+
+from logger import Logger
+from environments import Environments
+from retroconfig import Config
+
+# Only needed if we want to reinstate my own Kodi repository
+# from xbmcwrapper import XbmcWrapper
+# from helpers.languagehelper import LanguageHelper
+
+
+class EnvController:
+    """Controller class for getting all kinds of information about the
+    Kodi environment."""
+
+    __CurrentPlatform = None
+    __SQLiteEnabled = None
+
+    def __init__(self, logger=None):
+        """Class to determine platform depended stuff
+
+        :param Logger logger: a logger object that is used to log information to
+
+        """
+
+        self.logger = logger
+
+    @staticmethod
+    def update_local_addons_in_kodi():
```

Comment:
why are you triggering updates from within the addon?

Reply:
It was needed in the past. But not anymore. Removed it.